tools/omero/omero_filter.xml

```diff
@@ -0,0 +1,102 @@
+<tool id="omero_filter_id" name="OMERO IDs" version="@TOOL_VERSION@+galaxy@VERSION_SUFFIX@"
+      profile="20.01" license="MIT">
+    <description> with ezomero </description>
+    <macros>
+        <token name="@TOOL_VERSION@">5.18.0</token>
+        <token name="@VERSION_SUFFIX@">0</token>
+    </macros>
+    <xrefs>
+        <xref type="bio.tools">omero</xref>
+    </xrefs>
+    <requirements>
+        <requirement type="package" version="3.0.1">ezomero</requirement>
+        <!-- openjdk is needed: https://github.com/conda-forge/omero-py-feedstock/pull/16 -->
+        <requirement type="package" version="21.0.2">openjdk</requirement>
+    </requirements>
+    <command detect_errors="exit_code"><![CDATA[
+        python $__tool_directory__/omero_filter.py
+        --credential-file '$credentials'
+        --host $omero_host
+        --port $omero_port
+        --filter $filter
+        --value1 $value1
+        --id $did
+        --tsv_file $tsv
+        #if $filter == "KP"
+        --value2 $value2
+        #end if
+
+    ]]></command>
+    <configfiles>
+        <configfile name="credentials"><![CDATA[
+{
+    "username": "$__user__.extra_preferences.get('omero_account|username', $test_username)",
+    "password": "$__user__.extra_preferences.get('omero_account|password', $test_password)"
+}
+        ]]></configfile>
+    </configfiles>
+    <inputs>
+        <param name="omero_host" type="text" label="OMERO host URL">
+            <validator type="regex" message="Enter a valid host location, for example, your.omero.server">^[a-zA-Z0-9._-]*$</validator>
+            <validator type="expression" message="No two dots (..) allowed">'..' not in value</validator>
+        </param>
+        <param argument="omero_port" type="integer" optional="false" value="4064" label="OMERO port"/>
+        <param argument="filter" type="select" optional="true" label="Select filter">
+            <option value="filename">Project</option>
+            <option value="KP">Screen</option>
+            <option value="tag">Dataset</option>
+        </param>
+        <param argument="value1" type="text" optional="false" label="First filter criteria (i.e. filename)"/>
```

Comment:
All text parameters need validators/sanitizers. 

Text parameters can not be optional (since also the empty string is a string). You need a validator if you want to forbid an empty string. 

The `argument` attribute is most useful for public software (then users know which command line flag corresponds to which Galaxy tool parameter). Since this is an internal tool I would just use the `name` attribute. Or use `argument="--value1".